peerscout/keyword_extract/keyword_extract.py

```diff
@@ -138,37 +134,22 @@ def etl_keywords_get_latest_state(
         keyword_extract_config.existing_keywords_field,
         keyword_extractor=keyword_extractor
     )
-    processed_data = remove_keys(
-        data_with_extracted_keywords,
-        [
-            keyword_extract_config.state_timestamp_field,
-            keyword_extract_config.existing_keywords_field,
-            keyword_extract_config.text_field
-        ]
-    )
 
-    with NamedTemporaryFile() as named_temp_file:
-        write_to_file(processed_data, named_temp_file.name)
-        create_or_extend_table_schema(
-            keyword_extract_config.gcp_project,
-            keyword_extract_config.destination_dataset,
-            keyword_extract_config.destination_table,
-            named_temp_file.name
-        )
-        write_disposition = (
-            WriteDisposition.WRITE_APPEND
-            if keyword_extract_config.table_write_append
-            else WriteDisposition.WRITE_TRUNCATE
-        )
-        load_file_into_bq(
-            filename=named_temp_file.name,
-            table_name=keyword_extract_config.destination_table,
-            auto_detect_schema=True,
-            dataset_name=keyword_extract_config.destination_dataset,
-            write_mode=write_disposition,
-            project_name=keyword_extract_config.gcp_project
+    with TemporaryDirectory() as tempdir:
+        full_temp_file_location = Path.joinpath(
+            Path(tempdir, "downloaded_rows_data"))
+        latest_timestamp_iterator = (
```

Comment:
If we just assign the iterable and not use it (e.g. `keyword_extract_config.state_timestamp_field` is false), then the generator wouldn't actually perform any action as far as I understand. In this case it might be worth ensuring that we have a list.